src/states/adult/mod.rs

```diff
@@ -258,10 +261,10 @@ impl Adult {
 
         self.network_service_mut().remove_and_disconnect_all();
 
-        Transition::Relocate {
-            details,
+        Ok(Transition::Relocate {
+            details: signed_msg.clone(),
```

Comment:
is this clone really required?

Reply:
Not strictly, though it would make the code more complicated if it was not and it is a rare message.